.github/workflows/e2e-tests.yaml

```diff
@@ -0,0 +1,57 @@
+name: End-to-End Testing
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+
+  e2e-triggermesh:
+    name: Test TriggerMesh components
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.17'
+
+    - name: Go caches
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ github.job }}-${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ github.job }}-${{ runner.os }}-go-
+
+    - name: KinD Cluster
+      uses: container-tools/kind-action@v1
+      with:
+        knative_serving: v1.0.0
+        knative_kourier: v1.0.0
```

Comment:
should `knative_eventing` be also used?

Reply:
AFAIK none of the end-to-tests directly related to this repo rely on Eventing (direct connections only), so I wanted to save some resources.

But now that you mention it, it could be that the "addressable-resolver" role from Eventing is required to resolve `sink.ref`. Let me try to verify whether that is the case by running a real test.

Reply:
As I expected, TriggerMesh only requires the aforementioned ClusterRole to work, and none of the actual Eventing components. So instead of enabling the entire Eventing bundle (with the default Broker + Channel), I figured it was enough to do [this](https://github.com/triggermesh/triggermesh/compare/f5b3e8e25cc0a73dd3b465cdb414553aba270655..c66683529a0c2e21a913acaa7c80bc2a47654a0c).

Let me know whether that works for you.

Merry Christmas! 🎄

Reply:
sure thing!